src/mscorlib/src/System/Diagnostics/Eventing/XplatEventLogger.cs

```diff
@@ -122,12 +126,20 @@ private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnly
 
         internal protected  override void OnEventSourceCreated(EventSource eventSource)
         {
-            EnableEvents(eventSource, EventLevel.LogAlways, EventKeywords.All, null);
+            string eventSourceFilter = eventSourceNameFilter.Value;
+            if (String.IsNullOrEmpty(eventSourceFilter) || (eventSource.Name.IndexOf(eventSourceFilter, StringComparison.OrdinalIgnoreCase) >= 0))
```

Comment:
There isn't an overload for Contains that takes a StringComparison, I used IndexOf so it could be case insensitive.